contracts/credits/src/msg.rs

```diff
@@ -0,0 +1,154 @@
+    /// to burn owner's CNTRN tokens and mint NTRN tokens in 1:1 proportion certain amount for owner.
+    /// Used to skip vesting as a reward for participating in the lockdrop.
+    /// [Permissioned - Lockdrop address]
+    BurnFrom { owner: String, amount: Uint128 },
+    /// If authorized (only dao can call),
+    /// locks the NTRN tokens and mints CNTRN tokens in 1:1 amount
+    /// and adds to the dao balance.
+    /// [Permissioned - DAO] (Dao set in initialize func as cw20 minter)
+    Mint {},
+}
+
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    /// Returns the current vestings of the given address.
+    #[returns(WithdrawableAmountResponse)]
+    WithdrawableAmount { address: String },
+    /// Returns the amount that is left vested of the given address.
+    #[returns(VestedAmountResponse)]
+    VestedAmount { address: String },
+    /// Returns the current allocation of the given address.
+    #[returns(AllocationResponse)]
+    Allocation { address: String },
+    /// Returns the current balance of the given address, 0 if unset.
+    #[returns(cw20::BalanceResponse)]
+    Balance { address: String },
+    /// Returns the total supply at provided height, or current total supply if `height` is unset.
+    #[returns(TotalSupplyResponse)]
+    TotalSupplyAtHeight { height: Option<u64> },
+    /// Returns the balance of the given address at a given block height or current balance if `height` is unset.
+    /// Returns 0 if no balance found.
+    #[returns(cw20::BalanceResponse)]
+    BalanceAtHeight {
+        address: String,
+        height: Option<u64>,
+    },
+    /// Returns metadata on the contract - name, decimals, supply, etc.
+    #[returns(cw20::TokenInfoResponse)]
+    TokenInfo {},
+    /// Returns who can mint and the hard cap on maximum tokens after minting.
+    #[returns(cw20::MinterResponse)]
+    Minter {},
+    /// Returns how much spender can use from owner account, 0 if unset.
+    #[returns(cw20::AllowanceResponse)]
+    Allowance { owner: String, spender: String },
+    /// Returns all allowances this owner has approved. Supports pagination.
+    #[returns(cw20::AllAllowancesResponse)]
+    AllAllowances {
+        owner: String,
+        start_after: Option<String>,
+        limit: Option<u32>,
+    },
+    /// Returns all allowances this spender has been granted. Supports pagination.
+    #[returns(cw20::AllSpenderAllowancesResponse)]
+    AllSpenderAllowances {
+        spender: String,
+        start_after: Option<String>,
+        limit: Option<u32>,
+    },
+    /// Returns all accounts that have balances. Supports pagination.
+    #[returns(cw20::AllAccountsResponse)]
+    AllAccounts {
+        start_after: Option<String>,
+        limit: Option<u32>,
+    },
+    /// Returns current config of Credits contract
+    #[returns(ConfigResponse)]
+    Config {},
+}
+
+#[cw_serde]
+pub struct MigrateMsg {}
+
+#[cw_serde]
+pub struct ConfigResponse {
```

Comment:
is it needed? maybe simply use Config?

Reply:
usually in other contracts I've seen same wrap into response object, so did it in a consistent way)

Reply:
not sure tho, can remove if everybody agrees

Reply:
I'd use the same structure as long as the response is exactly the same as the config itself. imagine a situation when we need to modify a Config structure and add a new field to it. if we do so with a separate config response structure, we will probably forget to add this field to the response structure, and the config response will be outdated.